pyspark/bigdl/models/rnn/README.md

```diff
@@ -0,0 +1,108 @@
+
+
+## Prepare the Input Data
+You can download the Tiny Shakespeare Texts corpus from [here](https://raw.githubusercontent.com/karpathy/char-rnn/master/data/tinyshakespeare/input.txt).
+
+After downloading the text, separate it into train.txt and val.txt. In our case, we just select 80 percentage of the input to be train and remaining 20 percentage to be val.
+Please upload the directory will contains train.txt and val.txt to hdfs. The program will later read in the original text file from hdfs.
+```shell
+export LANG=en_US.UTF-8
+head -n 8000 input.txt > val.txt
+tail -n +8000 input.txt > train.txt
+```
+
+### Sample Text
+
+The input text may look as follows:
+
+```
+      First Citizen:
+      Before we proceed any further, hear me speak.
+
+      All:
+      Speak, speak.
+
+      First Citizen:
+      You are all resolved rather to die than to famish?
+```
+## Preprocessing
+
+The <code>sentences_split</code>, <code>sentence_tokenizer</code> use [NLTK Toolkit](http://www.nltk.org/). NLTK is included
+in created virtual env(see ${BigDL_HOME}/pyspark/python_package/README.md to see how to create virtual env). As NLTK expects english.pickle to tokenizer, we need download punkt and navigate into the folder for packaging
+```
+$python
+>>> import nltk
+>>> nltk.download('punkt')
+```
+then you can find nltk_data folder, zip file with tokenizer.zip#tokenizer
+``` 
+$ cd nltk_data/tokenizers/
+$ zip -r ../../tokenizers.zip *
+$ cd ../../
+```
+
+### Sample Sequence of Processed Data
+```
+      3998,3875,3690,3999
+      3998,3171,3958,2390,3832,3202,3855,3983,3883,3999
+      3998,3667,3999
+      3998,3151,3883,3999
+      3998,3875,3690,3999
+```
+
+## Train the Model
+Example command:
+```bash
+./dist/bin/bigdl.sh -- \
+
+BigDL_HOME=...
+SPARK_HOME=...
+MASTER=yarn
+PYTHON_API_PATH=${BigDL_HOME}/dist/lib/bigdl-VERSION-python-api.zip
+BigDL_JAR_PATH=${BigDL_HOME}/dist/lib/bigdl-VERSION-jar-with-dependencies.jar
+PYTHONPATH=${PYTHON_API_PATH}:$PYTHONPATH
+# set http_proxy if you need proxy to access internet
+http_proxy=...
+PYSPARK_DRIVER_PYTHON=./venv/bin/python PYSPARK_PYTHON=./venv.zip/venv/bin/python ${SPARK_HOME}/bin/spark-submit \
+       --master ${MASTER} \
+       --deploy-mode client \
+      --conf spark.executorEnv.http_proxy=${http_proxy} \
+      --driver-memory 10g  \
+      --executor-cores 1  \
+      --executor-memory 60g \
+       --py-files ${PYTHON_API_PATH} \
+       --properties-file ${BigDL_HOME}/dist/conf/spark-bigdl.conf \
+       --jars ${BigDL_JAR_PATH} \
+       --archives venv.zip,tokenizers.zip#tokenizers \
+       --conf spark.driver.extraClassPath=${BigDL_JAR_PATH} \
+       --conf spark.executor.extraClassPath=bigdl-version-jar-with-dependencies.jar \
+       --conf spark.yarn.appMasterEnv.NLTK_DATA=./ \
+       --num-executors 1 \
+       ${BigDL_HOME}/pyspark/dl/models/rnn/rnnexample.py --folder hdfs://almaren-node-001:9000/rnn/ --batchSize 12
```

Comment:
add a command for local integration test here ($BigDL/pyspark/test/local_integration/commands)?  we might enable it on jenkins pretty soon.